packages/ui-extensions/src/surfaces/admin/api/discount-function-settings/launch-options.ts

```diff
@@ -7,23 +7,69 @@ interface Metafield {
   type: string;
 }
 
-export enum DiscountClass {
+enum DiscountClass {
   Product = 'PRODUCT',
   Order = 'ORDER',
   Shipping = 'SHIPPING',
 }
 
-interface Discount {
-  /**
-   * the discount's gid
-   */
-  id: string;
+enum DiscountMethod {
+  Automatic = 'Automatic',
+  Code = 'Code',
+}
+
+enum PurchaseType {
+  OneTimePurchase = 'ONE_TIME_PURCHASE',
+  Subscription = 'SUBSCRIPTION',
+  Both = 'BOTH',
+}
+
+type Result<T> =
+  | {success: true; value: T}
+  | {success: false; error: ValidationError};
+
+interface ValidationError {
+  type: 'error';
+  message: string;
+  code: string;
+  issues?: {
+    message: string;
+    path: string[];
+  }[];
 }
 
 /**
  * The object that exposes the validation with its settings.
  */
 export interface DiscountFunctionSettingsData {
-  id: Discount;
+  id: string;
+  discountClasses: DiscountClass[];
   metafields: Metafield[];
 }
+
+/**
+ * Represents a reactive signal interface that provides both immediate value access and subscription-based updates. Enables real-time synchronization with changing data through the observer pattern.
+ */
+export interface ReadonlySignalLike<T> {
```

Comment:
Could you pull this in from a shared type? We have signals in other surface areas so there should be something you can use